Sources/SwiftGRPCNIO/HTTP1ToRawGRPCServerCodec.swift

```diff
@@ -120,17 +85,21 @@ extension HTTP1ToRawGRPCServerCodec: ChannelOutboundHandler {
     case .headers(let headers):
       //! FIXME: Should return a different version if we want to support pPRC.
       ctx.write(self.wrapOutboundOut(.head(HTTPResponseHead(version: .init(major: 2, minor: 0), status: .ok, headers: headers))), promise: promise)
-    case .message(var messageBytes):
-      // Write out a length-delimited message payload. See `channelRead` fpor the corresponding format.
-      var responseBuffer = ctx.channel.allocator.buffer(capacity: messageBytes.readableBytes + 5)
-      responseBuffer.write(integer: Int8(0))  // Compression flag: no compression
-      responseBuffer.write(integer: UInt32(messageBytes.readableBytes))
-      responseBuffer.write(buffer: &messageBytes)
-      ctx.write(self.wrapOutboundOut(.body(.byteBuffer(responseBuffer))), promise: promise)
+
+    case .message(let message):
+      do {
+        let responseBuffer = try messageWriter.write(allocator: ctx.channel.allocator, compression: .none, message: message)
+        ctx.write(self.wrapOutboundOut(.body(.byteBuffer(responseBuffer))), promise: promise)
+      } catch {
+        ctx.fireErrorCaught(error)
```

Comment:
Fail the promise? When can `messageWriter` throw?

Reply:
It throws if the compression mechanism is not supported. Given the mechanism will be determined beforehand, we can assure that it is supported at this point so we can remove the throwing.